wordpress-mobile/WordPress-Android
# Adding spent time in reader tracking

Fixes #10958

We are going to add some properties to the already application_closed event to trace some figures about time spent in reader.

Currently we have this in the above event:

> application_closed, Properties: {“last_visible_screen”:“Notifications”,“time_in_app”:25}

We are going to have something like this:

> application_closed, Properties: {“time_in_reader_filtered_list”:0,“time_in_reader_paged_post”:0,“time_in_main_reader”:6,“last_visible_screen”:“Notifications”,“time_in_app”:25}

where:
- `time_in_main_reader` : this is the seconds spent in the main reader (traced between onResume/onPause in ReaderPostListFragment when called with mIsTopLevel == true that is when it is used as main reader screen)
- `time_in_reader_filtered_list` : seconds in ReaderPostListFragment(NOT TopLevel) -> these are all the list of posts outside the main page of the reader (for example when you open a site list of post from the main reader OR from the reader settings > Followed Sites
- `time_in_reader_paged_post` : seconds in ReaderPostPagerActivity -> when opening a list of paged posts where you can swipe between single posts (for example from the list of posts in the main reader you can open a single post and swipe between them)

## To test
- Unit tests must pass
- Be sure you have tracking turned on
- you can observe logcat while making following steps, filtering with this `(TRACK READER|Tracked: application_closed)` and Regex active
- Open the vanilla flavor of the app
- Trigger the 3 above conditions then put the app in background
- Check that you get some message like 
`Tracked: application_closed, Properties: {"time_in_main_reader":2,"last_visible_screen":"Reader","time_in_reader_paged_post":0,"time_in_app":6,"time_in_reader_filtered_list":0}`
and the reported timings are somewhat you expect
- Smoke test the app selecting reader and other features and check the reader timing does not accumulate


PR submission checklist:
